File: 392. Is Subsequence/isSubsequence.py
class Solution:
    def isSubsequence(self, s: str, t: str) -> bool:
        #Use a two pointer approach, set both to 0
        i, j = 0, 0
        
        #While i and j are both in range...
        while i <len(s) and j<len(t):
            #If the letter at i is equal to the letter at j, increment i
            if s[i] == t[j]:
                i += 1
            #j gets incremented no matter the scenario so we don't need an else statement
            j += 1
        
        #When we get to the end, i should be equal to the length of s, otherwise no subsequence
        return i == len(s)
        
            

File: 392. Is Subsequence/isSubsequence.java
class Solution {
    public boolean isSubsequence(String s, String t) {
        //Throw in this edge case or we will get a security manager error
        if (s.equals("")){
            return true;
        }
        
        //Initialize i to 0, loop through the t string with j intialized to 0
        int i = 0;
        for (int j = 0; j<t.length(); j++){
            //If i char is equal to j char, increment i. J is automatically incremented no matter what
            if (s.charAt(i) == t.charAt(j)){
                i++;
            }
        //When we get to the end of the iterations, i should be equal to s.length
        if (i == s.length()){
            return true;
            }
        }
        return false; 
    }
}

